### DISCUSSION TITLE
Load only polyglot query engines present in the classpath

### DISCUSSION THREAD

**Discussion by vic0824** [ArcadeData/arcadedb#729] [Ideas]
Since v22.11.1, I have to carry 40MB of extra jars that I will never need, because ArcadeDB will not let me create a database via the Java API if I don't have js-21.2.0.jar, truffle-api-21.2.0.jar and icu4j-69.1.jar on the classpath.

I believe it would be much better if ArcadeDB loaded only the languages for which there is a corresponding library on the classpath, perhaps exploiting the server plugin mechanism, or the Java Service Provider Interface.

**Comment by lvca** [comment]
Starting from 21.11 we support custom functions that use GraalVM for Javascript and other languages. We already implement a pluggable mechanism for Gremlin,+Cypher, MongoDB, Redis, Postgres, and other plugins/languages, so we can totally to this.

The work is to define some hooks in the engine and let the custom function plugin use them.

**Comment by lvca** [comment]
With my last fix if any languages configured has missing jars, shouldn't cause an issue.
